src/main/kotlin/org/rust/ide/intentions/UnwrapToTryIntention.kt

```diff
@@ -0,0 +1,31 @@
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsMethodCallExpr
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsTryExpr
+import org.rust.lang.core.psi.util.parentOfType
+
+class UnwrapToTryIntention : RsElementBaseIntentionAction<RsMethodCallExpr>() {
+    override fun getText() = "Replace .unwrap() with try"
+    override fun getFamilyName() = text
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): RsMethodCallExpr? {
+        val methodCall = element
+            .parentOfType<RsMethodCallExpr>(strict = false) ?: return null
+        if (methodCall.identifier.textMatches("unwrap") &&
+            methodCall.typeArgumentList == null &&
+            methodCall.valueArgumentList.exprList.isEmpty()) {
+            return methodCall
+        }
+        return null
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: RsMethodCallExpr) {
+        val newPsiElement = RsPsiFactory(project).createExpression("a?")
+        (newPsiElement as? RsTryExpr)?.expr?.replace(ctx.expr) ?: return
```

Comment:
Sorry, I've conflated two thoughts here.

* Yeah, `?: error("Faild to foo expression ${bar.text}")` is better than `?: return`

* Instead of casting an expression to try expression, let's add a function to `RsPsiFactory` which construct `RsTryExpression`.